src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Insights/New-ArmApplicationInsightsWebTest.ps1

```diff
@@ -0,0 +1,122 @@
+function New-ArmApplicationInsightsWebTest {
+    [CmdletBinding(SupportsShouldProcess = $true)]
+    [OutputType("ApplicationInsightsWebTest")]
+    Param(
+        [Parameter(Mandatory, ValueFromPipeline)]
+        [ValidatePattern('^[a-zA-Z0-9-]*$')]
+        [string]
+        $Name,
+        [Parameter(Mandatory)]
+        [PSTypeName("ApplicationInsights")]
+        $ApplicationInsights,
+        [int]
+        [ValidateSet(300, 600, 900)]
+        $Frequency = 300,
+        [int]
+        [ValidateSet(30, 60, 90, 120)]
+        $Timeout = 30,
+        [string]
+        [ValidateSet("ping")]
+        $Kind = "ping",
+        [bool]
+        $RetryEnabled = $True,
+        [string[]]
+        [Parameter(Mandatory)]
+        [ValidateSet("emea-nl-ams-azr", "us-ca-sjc-azr", "emea-ru-msa-edge", "emea-se-sto-edge", "apac-sg-sin-azr", "us-tx-sn1-azr", "us-il-ch1-azr", "emea-gb-db3-azr", "apac-jp-kaw-edge", "emea-ch-zrh-edge", "emea-fr-pra-edge", "us-va-ash-azr", "apac-hk-hkn-azr", "us-fl-mia-edge", "latam-br-gru-edge", "emea-au-syd-edge")]
+        $Locations,
+        [string]
+        [Parameter(Mandatory)]
+        $Url,
+        [bool]
+        $ParseDependentRequests = $False,
+        [bool]
+        $FollowRedirects = $True,
+        [int]
+        [ValidateRange(200, 500)]
+        $ExpectedHttpStatusCode = 200,
+        [string]
+        $ExpectedResponseUrl = ""
+    )
+
+    If ($PSCmdlet.ShouldProcess("Creates a new Application Insights Web Test")) {
+        $ApplicationInsightsResourceId = $ApplicationInsights._ResourceId.Replace('[', '').Replace(']', '')
+        $ApplicationInsightsName = $ApplicationInsights.name
+        $WebTestId = [GUID]::NewGuid().ToString()
+        $WebRequestId = [GUID]::NewGuid().ToString()
```

Comment:
Non deterministic code will conflict with DSC no? If it doesn,t need to be a guid I suggest we use the arm template function [uniqueID()]